bluebutton-data-pipeline-app/src/test/java/gov/hhs/cms/bluebutton/datapipeline/app/S3ToFhirLoadAppIT.java

```diff
@@ -0,0 +1,425 @@
+package gov.hhs.cms.bluebutton.datapipeline.app;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import org.awaitility.Awaitility;
+import org.awaitility.Duration;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.AssumptionViolatedException;
+import org.junit.Test;
+
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.Bucket;
+
+import gov.hhs.cms.bluebutton.datapipeline.fhir.load.FhirTestUtilities;
+import gov.hhs.cms.bluebutton.datapipeline.rif.extract.s3.DataSetManifest;
+import gov.hhs.cms.bluebutton.datapipeline.rif.extract.s3.DataSetManifest.DataSetManifestEntry;
+import gov.hhs.cms.bluebutton.datapipeline.rif.extract.s3.DataSetTestUtilities;
+import gov.hhs.cms.bluebutton.datapipeline.rif.model.RifFileType;
+
+/**
+ * <p>
+ * Integration tests for {@link S3ToFhirLoadApp}.
+ * </p>
+ * <p>
+ * These tests require the application capsule JAR to be built and available.
```

Comment:
Two reasons, one that's practical and another that's philosophical:

1. Practical: Without writing the code to accommodate something different just for the tests, the only way to pass in the env vars configuration is via a separate process.
    * I don't think this is a compelling reason by itself, as design-for-test is important!
2. Philosophical: It's important to have some test coverage of the final end product -- the binary that we're actually going to be shipping. Any tests within the same JVM wouldn't verify the capsule is built correctly, or even that the runtime classpath is legit (as it would use the test classpath).
    * At a minimum, I'd want an automated smoketest of the final binary. This gives us that, plus some.
    * The process of getting this passing bears out this approach: I spent **a lot** of time fixing stuff that wouldn't have shown up if we were testing things in the same process. Getting the application to actually exit for errors and a `ctrl+c` was a bit of an adventure.